### PR TITLE
getBitmap() API test

### PR DESCRIPTION
**Pull request by rhyolight**
Contains mocked HTTP with httpretty. Also updated requirements to use "sure" assertion module and updated httpretty module.

### CODE REVIEW

**Comment by rhyolight** [comment]
@BoltzmannBrain

File: tests/unit/mock-api-data/terms-cat.json [hunk 1]
```diff
@@ -0,0 +1,342 @@
+[
+  {
+    "term": "cat",
```
Comment (BoltzmannBrain) [inline tests/unit/mock-api-data/terms-cat.json:3]: @rhyolight I did not think at all of including a mock data file... this answers a lot of my questions about mocking out API calls.

Comment (rhyolight) [inline tests/unit/mock-api-data/terms-cat.json:3]: The data is easy to get from http://www.cortical.io/resources_apidocumentation.html

File: tests/unit/client_test.py [hunk 2]
```diff
@@ -1,15 +1,73 @@
...
+    # Act.
+    # --------
+    bitmap = client.getBitmap("cat")
+
+    # Assert.
+    # --------
+    # Check result object.
+    (bitmap).should.have.key("term").being.equal("cat")
+    (bitmap).should.have.key("fingerprint").being.a("dict")
+    (bitmap["fingerprint"]).should.have.key("positions").being.a("list")
```
Comment (rhyolight) [inline tests/unit/client_test.py:130]: You might want to add further assertions here, but the main idea is to assert the data structure you get back from the function under test has been converted properly from the mock API JSON response.

**Comment by rhyolight** [comment]
@BoltzmannBrain This is ready to merge if you are comfortable with it. I'm happy to discuss any of the test code here with you.

File: tests/unit/client_test.py [hunk 3]
```diff
@@ -1,15 +1,143 @@
 import os
+
+import sure
```
Comment (rhyolight) [inline tests/unit/client_test.py:3]: This is the import that magically gives you the [BDD-style assertions](http://falcao.it/sure/). You might hate them, in which case I'm happy swapping them out with more standard assertions. I find them very readable.

Comment (BoltzmannBrain) [inline tests/unit/client_test.py:3]: I wouldn't say I hate them, but they're certainly new to me and seem unconventional. Kind of cool actually. Besides readability are there advantages over the [unittest assert methods](https://docs.python.org/2/library/unittest.html#assert-methods)?

Comment (rhyolight) [inline tests/unit/client_test.py:3]: I think the main advantage is readability, which is a big benefit for unit
test code, which can get really obscured really fast. I definitely would
NOT use something like "sure" in production code.

---

Matt Taylor
OS Community Flag-Bearer
Numenta

File: cortipy/cortical_client.py [hunk 4]
```diff
@@ -66,6 +66,7 @@ def __init__(self,
                baseUrl=DEFAULT_BASE_URL,
                retina=DEFAULT_RETINA,
                cacheDir=DEFAULT_CACHE_DIR,
+               useCache=True,
```
Comment (BoltzmannBrain) [inline cortipy/cortical_client.py:69]: :+1:

File: tests/unit/client_test.py (hunk 3 above)
Comment (BoltzmannBrain) [inline tests/unit/client_test.py:3]: Sure I'll try sure (hah), but I may decide to swap them out later for standard assertions.

**Pull request merged**